data-prepper-plugins/log-generator-source/src/test/java/com/amazon/dataprepper/plugins/source/loggenerator/LogGeneratorSourceTest.java

```diff
@@ -0,0 +1,147 @@
+    private PluginMetrics pluginMetrics;
+    private PluginFactory pluginFactory;
+    private LogGeneratorSource logGeneratorSource;
+
+    @BeforeEach
+    public void setup() {
+        sourceConfig = mock(LogGeneratorSourceConfig.class);
+        pluginMetrics = mock(PluginMetrics.class);
+        pluginFactory = mock(PluginFactory.class);
+
+        PluginModel mockLogPluginModel = mock(PluginModel.class);
+        when(sourceConfig.getLogType()).thenReturn(mockLogPluginModel);
+
+        when(mockLogPluginModel.getPluginName()).thenReturn(
+                "TestPluginName"
+        );
+        Map<String, Object> emptyMap = new HashMap<>();
+        when(mockLogPluginModel.getPluginSettings()).thenReturn(
+                emptyMap
+        );
+        LogTypeGenerator commonApacheLogTypeGenerator = mock(LogTypeGenerator.class);
+
+
+        when(pluginFactory.loadPlugin(any(), any(PluginSetting.class))).thenReturn(commonApacheLogTypeGenerator);
+        String hardcodedEvent = "127.0.0.1 user pwd [date] \"GET /list HTTP/1.1\" 404 4043";
+        Event stubbedJackson = JacksonEvent.fromMessage(hardcodedEvent);
+        lenient().when(commonApacheLogTypeGenerator.generateEvent()).thenReturn(stubbedJackson); // JacksonEvent type
+
+        logGeneratorSource = new LogGeneratorSource(sourceConfig, pluginMetrics, pluginFactory);
+    }
+
+    @Test
+    void GIVEN_logGeneratorSourceAndBlockingBuffer_WHEN_noLimit_THEN_keepsWritingToBufferUntilStopped()
+            throws InterruptedException, TimeoutException {
+        BlockingBuffer<Record<Event>> spyBuffer = spy(new BlockingBuffer<Record<Event>>("SamplePipeline"));
+
+        lenient().when(sourceConfig.getInterval()).thenReturn(Duration.ofSeconds(1)); // interval of 1 second
+        lenient().when(sourceConfig.getCount()).thenReturn(0); // no limit to log count
+
+        logGeneratorSource.start(spyBuffer);
+        Thread.sleep(1500);
+
+        verify(spyBuffer, atLeast(1)).write(any(Record.class), anyInt());
+        Thread.sleep(700);
+        verify(spyBuffer, atLeast(2)).write(any(Record.class), anyInt());
+        logGeneratorSource.stop();
+    }
+
+    @Test
+    void GIVEN_logGeneratorSourceAndBlockingBuffer_WHEN_reachedLimit_THEN_stopsWritingToBuffer()
+            throws InterruptedException, TimeoutException {
+        BlockingBuffer<Record<Event>> spyBuffer = spy(new BlockingBuffer<Record<Event>>("SamplePipeline"));
+
+        lenient().when(sourceConfig.getInterval()).thenReturn(Duration.ofSeconds(1)); // interval of 1 second
+        lenient().when(sourceConfig.getCount()).thenReturn(1); // max log count of 1 in logGeneratorSource
+
+        logGeneratorSource.start(spyBuffer);
+        assertEquals(spyBuffer.isEmpty(), true);
+        Thread.sleep(1100);
+
+        verify(spyBuffer, times(1)).write(any(Record.class), anyInt());
+
+        Thread.sleep(1000);
+        verify(spyBuffer, times(1)).write(any(Record.class), anyInt());
+        logGeneratorSource.stop();
```

Comment:
You can move this call to an `@AfterEach` method to avoid repetition

Reply:
Are you referring to the `logGeneratorSource.stop()` line here? The `verify` call verifies different behavior in the two tests so I don't think it should be moved to an `@AfterEach` method.

Reply:
Yes I'm referring to the `logGeneratorSource.stop()` line which is in both of the unit tests